src/gui/DetailsWidget.cpp

```diff
@@ -75,9 +87,10 @@ void DetailsWidget::getSelectedEntry(Entry* selectedEntry)
 
     m_ui->stackedWidget->setCurrentIndex(EntryPreview);
 
-    if (m_ui->tabWidget->count() < 4) {
-        m_ui->tabWidget->insertTab(static_cast<int>(AttributesTab), m_attributesWidget, "Attributes");
-        m_ui->tabWidget->insertTab(static_cast<int>(AutotypeTab), m_autotypeWidget, "Autotype");
+    if (m_ui->tabWidget->count() < 5) {
```

Comment:
I completely agree with you about the magic numbers. I think it is necessary to refactor DetailsWidget in another pull request. I can do that after this changes will be merged. If you don't mind.

Reply:
I think it would be enough to simply clear out the widget and add stuff as needed (I think it's unlikely that this would impact performance massively).